### PR TITLE
enact def + = newline rules

### PR DESCRIPTION
**Pull request by V-FEXrt**
Enacts the rules for newlining defs as discussed in chat.

Specially
- Same rules as current formatter (the first bikeshed PR) plus
- Top level = always newlines
- except no newlines for single top level literals/identifiers

### CODE REVIEW

File: tools/wake-format/emitter.cpp [hunk 1]
```diff
@@ -262,11 +287,16 @@ auto Emitter::rhs_fmt() {
...
       return c->has_newline() && c->last_width() == 0;
    }, full_fmt)
-   // If the RHS has a leading comment then we must use the full_fmt
+    // If the RHS has a leading comment then we must use the full_fmt
    .pred([this](const wcl::doc_builder& builder, ctx_t ctx, const CSTElement& node,
             const token_traits_map_t& traits) {
       return count_leading_newlines(token_traits, node) > 0;
    }, full_fmt)
+    // Always newline when requested unless the thing to be formatted is a "single literal".
+    // Used for top-level defs and top-level "constatn" defs
```
Comment (ag-eitilt) [inline tools/wake-format/emitter.cpp:296]:
```suggestion
    // Used for top-level defs and top-level "constant" defs
```

File: tools/wake-format/emitter.h [hunk 2]
```diff
@@ -70,6 +72,9 @@ class Emitter {
   // Marks all node elements that have had their formatting disabled
   void mark_no_format_nodes(CSTElement node);
 
+  // Marks top level nodes. Any node on 'top level' will have top level = true
+  void mark_top_level_notes(CSTElement node);
```
Comment (ag-eitilt) [inline tools/wake-format/emitter.h:76]:
```suggestion
  void mark_top_level_nodes(CSTElement node);
```
...and its uses.

**Review by ag-eitilt — APPROVED** [review]
Couple spelling fixes, but otherwise looks good!

File: tools/wake-format/emitter.h (hunk 2 above)
Comment (V-FEXrt) [reply tools/wake-format/emitter.h]: nice catches! Thanks!

**Pull request merged**